utils/modular_model_converter.py

```diff
@@ -808,322 +1005,261 @@ def visit_ImportFrom(self, node: cst.ImportFrom) -> None:
-        `True`. Return `False` otherwise.
+        # Now, visit every model-specific files found in the imports, and merge their dependencies
+        self.visited_modules = {}
+        self.renamers = {}
+        for file, module in self.model_specific_modules.items():
+            file_model_name = re.search(r"models\.\w*?\.\w*?_(\S*)", file).groups()[0]
+            renamer = ReplaceNameTransformer(
+                file_model_name, self.model_name, self.given_old_name, self.given_new_name
+            )
+            renamed_module = module.visit(renamer)
+            self.visited_modules[file] = ModelFileMapper.visit_and_merge_dependencies(
+                renamed_module,
+                self.functions,
+                self.function_call_dependency_mapping,
+                self.assignments,
+                self.start_lines,
+            )
+            # We record it so that we can rename classes later the exact same way
+            self.renamers[file] = renamer
+
+        # In turn, we need to add the imported functions/assignments to the dependencies of the modular mapper, using the
+        # definitions found in the visited files
+        self.merge_model_specific_imports(self.visited_modules)
+
+        # Re-assign all nodes
+        self.global_nodes = {**self.assignments, **self.classes, **self.functions}
+
+    def merge_model_specific_imports(self, visited_modules):
+        """Merge the model-specific imported functions and assignments to the modular nodes and dependency graph,
+        based on the visited files."""
+        self.start_lines_file_mapping = {}
+        self.added_objects_file_mapping = {}
+        for object_name, file in self.model_specific_imported_objects.items():
+            visited_module = visited_modules[file]
+            self.start_lines_file_mapping[file] = visited_module.start_lines
+            # Add functions and their dependencies
+            if object_name in visited_module.functions and object_name not in self.functions:
+                self.functions[object_name] = visited_module.functions[object_name]
+                self.added_objects_file_mapping[object_name] = file
+                dependencies = visited_module.function_call_recursive_dependency_mapping.get(object_name, None)
+                if dependencies is not None:
+                    self.function_call_recursive_dependency_mapping[object_name] = dependencies
+                    for dep in dependencies:
+                        self.added_objects_file_mapping[dep] = file
+                        self.functions[dep] = visited_module.global_nodes[dep]
+
+            # Add assignments
+            elif object_name in visited_module.assignments and object_name not in self.assignments:
+                self.added_objects_file_mapping[object_name] = file
+                self.assignments[object_name] = visited_module.assignments[object_name]
+
+    def compute_relative_order(self, missing_dependencies: set) -> dict[str, int]:
```

Comment:
does this need to be in this visiter? probably not no?

Reply:
As the goal anf functionality is the same but the logic is different depending on the type of `Mapper`, I think it makes sense to keep it in the visitors. (It uses a lot of `self` internal variables as well)